src/error.rs

```diff
@@ -101,6 +103,10 @@ pub enum Error {
 	LiquiditySourceUnavailable,
 	/// The given operation failed due to the LSP's required opening fee being too high.
 	LiquidityFeeTooHigh,
+	/// Returned when trying to open an announced channel with a peer. This
+	/// error occurs when a [`crate::Node`'s] alias or listening addresses
+	/// are unconfigured.
+	OpenAnnouncedChannelFailed,
```

Comment:
I decided to use the existing error variants as suggested, i.e. `InvalidNodeAlias` and `InvalidSocketAddress`. To ensure that we are selecting the correct variant and that we properly inform callers about why opening a channel might fail, I introduced two enums (see snippet below) so that we can attach more context to channel announcement.

https://github.com/enigbe/ldk-node/blob/4fd1cb8ed98c5875627e31d9c7cbf3c5e0fe8d6b/src/config.rs#L279-L297

https://github.com/enigbe/ldk-node/blob/4fd1cb8ed98c5875627e31d9c7cbf3c5e0fe8d6b/src/config.rs#L304C1-L320C2